[Lin Yuan Xun] iP

YuanXun9834 opened the pull request:
# **Duke Application**

>"Your mind is for having ideas, not holding them." - David Allen

The Duke application helps frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPERFAST to use!

All you need to do is,

1.download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html)
2.double click it
3.add your tasks
4.let it manage your tasks for you :+1:

And it is **FREE**!

Features:

- [x] Managing Tasks
- [x] Managing Events
- [ ] Reminders! (Coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Complete Level-1

Commit:
Complete Level-2

Commit:
complete Level-3

Commit:
Complete Level-4

Commit:
Completed A-TextUiTesting

Commit:
Completed A-Exceptions

Commit:
Completed Level-6

Commit:
Add FileReadWrite.java

Commit:
Update date representation in Deadlines and Events

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

Commit:
Add classes to handle Ui, Task List, parsing of input logic

Commit:
Add classes into duke package

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add junit tests for deadline and todo class

Commit:
Add Javadocs

Commit:
Modified to meet Coding standards

Commit:
Add find function to find tasks

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/EmptyArgException.java
#	src/main/java/duke/Event.java
#	src/main/java/duke/FileReadWrite.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Todo.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/UnknownInputException.java

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

Commit:
Added GUI to Duke

WillCWX commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,34 @@
+ */
+public class Duke {
+    private TaskList taskList;
+    private Ui ui;
```
The variable `ui` does not appear to be used. Perhaps you should delete it if you don't need it anymore.

WillCWX commented on src/main/java/duke/FileReadWrite.java:
```diff
@@ -0,0 +1,139 @@
+        if (!file.exists()) {
+            file.mkdir();
+        }
+        Scanner scanner = new Scanner(file);
```
It seems that the `Scanner scanner` was never closed and can cause a resource leak. Maybe you can close the scanner right before you `return taskList`?

This occurs in the following as well:
 ```
    public static void writeMark(int i, Task t) throws IOException {
        File file = new File(FILE_PATH);
        Scanner scanner = new Scanner(file);
 ```
 ```
     public static void writeUnmark(int i, Task t) throws IOException {
        File file = new File(FILE_PATH);
        Scanner scanner = new Scanner(file);
 ```
 ```
     public static void writeDelete(int i) throws IOException {
        File file = new File(FILE_PATH);
        Scanner scanner = new Scanner(file);
 ```

WillCWX commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,122 @@
+     * @param input input string from user read from the Ui class
+     * @return boolean return false if the user indicates the end of current session
+     */
+
```
This extra space can be deleted

WillCWX commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+     * Function that prints all the tasks currently in the TaskList.
+     *
+     */
+
```
> No blank line between the documentation block and the method/class

This blank line is not needed according to [cs2103t](https://se-education.org/guides/conventions/java/index.html#comments)

This appears at the following as well:
```
     */

    public Task markTask(int index) {
```
```
     */

    public Task unmarkTask(int index) {
```
```
     */

    public Task delete(int index) {
```

WillCWX commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+        super();
+    }
+
+    /**
+     * Function that prints all the tasks currently in the TaskList.
+     *
+     */
```
Your JavaDoc is missing a `@return` tag and statement. Perhaps you could write it like this:
```suggestion
    /**
     * Function that prints all the tasks currently in the TaskList.
     *
     * @return The string containing all the tasks currently in the TaskList.
     */
```

WillCWX commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,82 @@
+        return output;
+    }
+
+    public String findList(String input) {
```
Your `findList` function is missing a JavaDoc.

WillCWX commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,26 @@
+ * Todo extends the Task class and represents a simple task description
+ * without any information associated with it
+ */
+
```
This line can be deleted.

WillCWX commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,115 @@
+    /**
+     * Message for un-mark operation
+     */
+    public static String printUnmark(TaskList taskList, int i) {
+        return "Alright! I've unmarked this task :(\n" + taskList.get(i - 1);
+    }
+
+    /**
+     * Message for mark operation
+     */
+    public static String printMark(TaskList taskList, int i) {
+        return "Alright! I've marked this task as done :) !\n" + taskList.get(i - 1);
+    }
+
+    /**
+     * Message for delete operation
+     */
+    public static String printDelete(Task task, int i, int size) {
+        String s = "OK! I've deleted this task :)\n";
+        s = s + task;
+        s = s + String.format("Now you have %d tasks in the list", size - 1);
+        return s;
+    }
+
+    /**
+     * Message for list operation
+     */
+    public static String printListCommand(TaskList taskList) {
+        return "Here are the remaining tasks in your list:\n" + taskList.printList();
+    }
+
+    /**
+     * Message for add task operation
+     */
+    public static String printAddTask(Task task, int size) {
+        String s = "Roger! I've added this task to the list:\n";
+        s = s + task + "\n";
+        s = s + String.format("Now you have %d tasks left in the list", size);
+        return s;
+    }
```
It appears that you are missing `@return` tags for all of these JavaDocs,

WillCWX commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,115 @@
+
+
+
+    public static String printFindList(TaskList taskList, String input) {
```
You are missing a JavaDoc for this static method.

WillCWX commented on src/main/java/duke/UnknownInputException.java:
```diff
@@ -0,0 +1,7 @@
+package duke;
+
+public class UnknownInputException extends DukeException {
```
This public class needs a JavaDoc as well.

WillCWX commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,14 @@
+package duke;
```
It appears that you are mssing a newline at the end of this file.

WillCWX reviewed (changes requested):
Well done on having multiple different kinds of `DukeException` that can be thrown to help with error handling.
However, there seems to be some style issues and JavaDoc missing. These can be easily identified by you once you finish [A-Checkstyle](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html#a-checkstyle) and perhaps some can be easily corrected by your IDE as well.

Commit:
Add Assertions
Assumptions about correctness of input were made in previous version of code.
Assertions made to ensure and document assumptions that should hold at certain points in the code
Assert statements are added to parse method since the method makes some assumptions about the input

Commit:
Refactor methods that were lacking in quality
Changed parseInput method to not have nested if else blocks

YuanXun9834 committed:
Merge pull request #1 from YuanXun9834/branch-A-Assertions

Add Assertions

Commit:
Merge branch 'branch-A-CodeQuality'

Commit:
Added Priority feature to the tasks,
Changed representation of task list in duke.txt to reflect priorities
Also changed string representation of all tasks

cylithegreat commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,45 @@
+        String s = "E___";
+        String d = this.getDone() ? "✓" : "X";
+        String p = this.getPriority() + "___";
+        return s + d + "___" + p + this.getTask() + "___" + this.getDate1() + "___" + this.getDate2();
```
I like that you remember to leave a space between the operands and the operators, great readability!

cylithegreat commented on src/main/java/duke/FileReadWrite.java:
```diff
@@ -0,0 +1,164 @@
+        lines.remove(i - 1);
+        t.unmark();
+        lines.add(i - 1, t.summary());
+        FileWriter fileWriter = new FileWriter(FILE_PATH);
```
Nice use of all-capitalized name for FILE_PATH 👍

cylithegreat commented on src/main/java/duke/Launcher.java:
```diff
@@ -0,0 +1,9 @@
+package duke;
+
+import javafx.application.Application;
+
+public class Launcher {
```
I like the line breaks in between, looks neat.

cylithegreat commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
```
I like that you arrange the import statements in alphabetical orders.

cylithegreat commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
```
Perhaps there is a typo here? Could the functional be named as valueProperty instead?

cylithegreat commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+        if (input == "bye") {
+            Platform.exit();
+        }
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(input, userImage),
+                DialogBox.getDukeDialog(response, dukeImage)
+        );
```
I appreciate your use of new line and spaces to separate a long line!

cylithegreat commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+     *
+     * @param index Index of task user wish to mark.
+     * @return Task that has been marked done.
+     */
+
+    public Task markTask(int index) {
+
```
Perhaps this line break could be deleted?

cylithegreat commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+        return output;
+    }
+
+    /**
+     * Marks a task in the Task List at index as done.
+     *
+     * @param index Index of task user wish to mark.
+     * @return Task that has been marked done.
+     */
```
I noticed that some of the function documentation starts right above the function, however some has a one line spacing in between. Perhaps you could consider standardizing them?

cylithegreat reviewed (commented):
I like how neat your codes are in general, they are easy to read and understand. 💯 
Just some very minor cosmetic changes recommended, as indicated in the individual comments.
Great job!

Commit:
Changed user.png as well as user guide

YuanXun9834 committed:
Add files via upload
